Primeiro PR Tutorial

VicAzv opened the pull request:
Primeiro e segundo tutorial concluído

IsadoraRFMuniz committed:
Primeiro PR Tutorial

desouzareis commented on hello-haplotype-caller.wdl:
```diff
@@ -0,0 +1,47 @@
+        File inputBAM
+        File bamIndex
+    }
+    call HaplotypeCaller{
```
Adicionar uma linha vazia separando as sessões. https://github.com/Varstation/bioinfo-help/wiki/Guia-de-estilos-de-WDL#linhas-vazias

desouzareis commented on hello-haplotype-caller.wdl:
```diff
@@ -0,0 +1,47 @@
+
+workflow HelloHaplotypeCaller {
+    input {
+        File GATK
```
Variáveis são nomeadas em lowerCamelCase. Ficaria
gatk
refFasta
refIndex
refDict...

desouzareis commented on hello-haplotype-caller.wdl:
```diff
@@ -0,0 +1,47 @@
+    output {
+        File rawVCF = "${sampleName}.raw.indels.snps.vcf"
+    }
+}
```
Seria importante adicionar o parameter_meta para documentação dos inputs e outputs. https://github.com/Varstation/bioinfo-help/wiki/Guia-de-estilos-de-WDL#sess%C3%A3o-parameter_meta

desouzareis commented on hello-haplotype-caller.wdl:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    command {
+        java -jar ${GATK} \
```
Utilizar  placeholder ~{...} para variáveis e não ${...}
https://github.com/Varstation/bioinfo-help/wiki/Guia-de-estilos-de-WDL#sess%C3%A3o-command

desouzareis commented on hello-haplotype-caller.wdl:
```diff
@@ -0,0 +1,47 @@
+        File bamIndex
+    }
+
+    command {
```
Utilizar a sintaxe de heredoc <<<>>> para o bloco de comando
https://github.com/Varstation/bioinfo-help/wiki/Guia-de-estilos-de-WDL#sess%C3%A3o-command

desouzareis commented on helloHaplotypeCaller_inputs.json:
```diff
@@ -0,0 +1,9 @@
+{
```
criar um pasta inputs para adicionar todos os inputs.json

desouzareis commented on simple-variant-selection.wdl:
```diff
@@ -0,0 +1,97 @@
+version 1.0
+
+workflow Tutorial_2 {
```
O nome do workflow poderia seguir o nome do arquivo `SimpleVariantSelection`

desouzareis commented on simple-variant-selection.wdl:
```diff
@@ -0,0 +1,97 @@
+workflow Tutorial_2 {
+    input {
+        File GATK
+        File RefFasta
```
Variáveis são lowerCamelCase

desouzareis commented on simple-variant-selection.wdl:
```diff
@@ -0,0 +1,97 @@
+            RefDict = RefDict,
+            sampleName = sampleName,
+            type = "SNP", 
+                rawVCF = Testando
```
Possui 1 recuo a mais https://github.com/Varstation/bioinfo-help/wiki/Guia-de-estilos-de-WDL#recuo

desouzareis commented on simple-variant-selection.wdl:
```diff
@@ -0,0 +1,97 @@
+            RefDict = RefDict,
+            sampleName = sampleName,
+            type = "INDEL", 
+                rawVCF = Testando 
```
Recuo a mais https://github.com/Varstation/bioinfo-help/wiki/Guia-de-estilos-de-WDL#recuo

desouzareis commented on simple-variant-selection.wdl:
```diff
@@ -0,0 +1,97 @@
+            -V ${rawVCF} \
+            -select-type ${type} \
+            -O ${sampleName}_raw.${type}.vcf
+  }
```
Recuo fora da indentação.

desouzareis commented on simple-variant-selection.wdl:
```diff
@@ -0,0 +1,97 @@
+        File bamIndex
+    }
+
+    command {
```
Utilizar A sintaxe de heredoc <<<>>> para o bloco de comando e placeholder ~{...} para variáveis

desouzareis commented on simple-variant-selection.wdl:
```diff
@@ -0,0 +1,97 @@
+        File rawVCF
+    }
+
+    command {
```
Utilizar A sintaxe de heredoc <<<>>> para o bloco de comando e placeholder ~{...} para variáveis

desouzareis reviewed (changes requested):
Boa Victor, segue sugestões para adoção ao nosso guia de estilos.

VicAzv committed:
Update para lowerCamelCase

VicAzv committed:
Update de todos os erros referentes ao guia

Separação de sessão, correção do heredoc e placeholder

VicAzv committed:
Update de acordo com o guia de estilos

O nome do workflow foi corrigido, as váriáveis foram colocadas no padrão lowerCamelCase, os comentários de recuo a mais confirmei que estão com 4 espaçamentos que é o padrão do guia, a sintaxes de heredoc e placeholder foram corrigidas, identação corrigida e foi retirado a última linha com comentário desnecessário.

IsadoraRFMuniz committed:
Pasta Inputs

IsadoraRFMuniz committed:
Merge branch 'Tutorial-1-e-2' of https://github.com/VicAzv/Tutorial-Weedle into Tutorial-1-e-2

VicAzv committed:
Mudança do nome do workflow

Adição de um nome que faça mais sentido para o workflow

desouzareis commented on simple-variant-selection.wdl:
```diff
@@ -0,0 +1,97 @@
+            -V ~{rawVcf} \
+            -select-type ~{type} \
+            -O ~{sampleName}_raw.~{type}.vcf
+        >>>
```
Esta linha possui um recuo a mais, deve finaliza na mesma coluna que o command

desouzareis commented on simple-variant-selection.wdl:
```diff
@@ -0,0 +1,97 @@
+            -select-type ~{type} \
+            -O ~{sampleName}_raw.~{type}.vcf
+        >>>
+}
```
Falta um recuo.

desouzareis commented on Inputs/helloHaplotypeCaller_inputs.json:
```diff
@@ -0,0 +1,9 @@
+{
```
O nome da pasta deveria ser com letra minúscula. 
O nome do arquivo deve seguir o mesmo nome do workflow.

desouzareis reviewed (changes requested):
Boa, solicitei pequenas mudanças. Mas está muito bem no fluxo de code review!

IsadoraRFMuniz committed:
Mudanças sugeridas no Github

IsadoraRFMuniz committed:
Alterações no nome dos inputs e variáveis

desouzareis reviewed (approved):
Top! PR aprovado, pode fazer o merge!
